selfdrive/car/tesla/teslacan.py

```diff
@@ -86,9 +41,24 @@ def create_longitudinal_commands(self, acc_state, speed, min_accel, max_accel, c
       "DAS_controlCounter": cnt,
       "DAS_controlChecksum": 0,
     }
+    data = self.packer.make_can_msg("DAS_control", CANBUS.party, values)[2]
+    values["DAS_controlChecksum"] = self.checksum(0x2b9, data[:7])
+    return self.packer.make_can_msg("DAS_control", CANBUS.party, values)
+
+  def right_stalk_press(self, counter, position):
+    # TODO: Implement CRC checksum instead of lookup table.
```

Comment:
would be _really_ nice to fix this

Reply:
I have been working on this for a long time. I have not been able to work out the calculation using any method. I have tried CRC-beagle, revng. I tried to calculate it manually, but without success.

Perhaps there is a clever mind out there who can solve this?

Reply:
@gregjhogan interested in trying? can upload a binary into ghidra if you want :)

Reply:
@lukasloetkolben  have you tried including message id before message bytes when doing reverse eng?

Reply:
Here is the algorithm for`SCCM_rightStalk` tested on a few messages:
```python
import crcmod

msgs = [
  b'\x8c\x0b\x00',
  b'\x31\x0e\x00',
  b'\x44\x01\x00',
]

right_stalk_init_val = [0x7C, 0xB6, 0xF0, 0x2F, 0x69, 0xA3, 0xDD, 0x1C, 0x56, 0x90, 0xCA, 0x09, 0x43, 0x7D, 0xB7, 0xF1]

for msg in msgs:
  cntr = msg[1] & 0xF
  crc1_func = crcmod.mkCrcFun(0x12f, initCrc=0x00, xorOut=0xFF, rev=False)
  crc1 = crc1_func(msg[1:]) & 0xFF
  crc2_func = crcmod.mkCrcFun(0x12f, initCrc=crc1, xorOut=0xFF, rev=False)
  crc2 = crc2_func(bytes([right_stalk_init_val[cntr]])) & 0xFF
  print(f'message: 0x{msg.hex()} -> crc: {crc2:02X} = {"GOOD" if crc2 == msg[0] else "BAD"}')
```

Reply:
just tested it on some more data. I will implement it later. thank you!